editor/scene_tree_dock.cpp

```diff
@@ -1682,6 +1682,30 @@ bool SceneTreeDock::_check_node_path_recursive(Node *p_root_node, Variant &r_var
 			}
 		} break;
 
+		case Variant::OBJECT: {
+			Resource *resource = Object::cast_to<Resource>(r_variant);
+			if (resource) {
```

Comment:
You could use early break instead:
```suggestion
			if (!resource) {
				break;
			}
```